metrics.go

```diff
@@ -198,25 +203,26 @@ func (se *statsExporter) createMetricDescriptorFromMetric(ctx context.Context, m
 		return nil
 	}
 
+	if builtinMetric(se.metricTypeFromProto(name)) {
+		se.metricDescriptors[name] = true
+		return nil
+	}
+
 	// Otherwise, we encountered a cache-miss and
 	// should create the metric descriptor remotely.
 	inMD, err := se.metricToMpbMetricDescriptor(metric)
 	if err != nil {
 		return err
 	}
 
-	if builtinMetric(inMD.Type) {
+	cmrdesc := &monitoringpb.CreateMetricDescriptorRequest{
```

Comment:
Consider making a helper function for this few lines so that `metrics`, `metrics_proto` and `stats` can reuse.

Reply:
Done.